### PR TITLE
fix: ensure time parts in parsed time object are numeric

### LINKED ISSUE #5101: Milliseconds in TimePicker are not properly displayed
### Description

When setting a _TimePicker_'s step to `Duration.ofMillis(1)` and setting its value to a time which millisecond value is less than 100 milliseconds, the UI displays "000" for milliseconds instead of the provided milliseconds.
Note: The `getValue()` method returns the correct value.

### Expected outcome

The provided milliseconds should be displayed.

### Minimal reproducible example

```java
public class MainView extends VerticalLayout {
    public MainView() {
        TimePicker wrongTimePicker1 = createTimePicker("00:00:00.001");
        TimePicker wrongTimePicker2 = createTimePicker("00:00:00.010");
        TimePicker wrongTimePicker3 = createTimePicker("00:00:00.099");
        TimePicker correctTimePicker1 = createTimePicker("00:00:00.100");
        TimePicker correctTimePicker2 = createTimePicker("00:00:00.101");

        add(wrongTimePicker1, wrongTimePicker2, wrongTimePicker3, correctTimePicker1, correctTimePicker2);
    }

    TimePicker createTimePicker(String value) {
        TimePicker timePicker = new TimePicker();
        timePicker.setStep(Duration.ofMillis(1));
        timePicker.setValue(LocalTime.parse(value, DateTimeFormatter.ofPattern("HH:mm:ss.SSS")));
        System.out.println(timePicker.getValue());
        return timePicker;
    }

}
```

### Steps to reproduce

1. Implement a view based on the snippet above.
2. Open the UI.
3. Notice that for the first 3 _TimePickers_ the displayed value is incorrect, even though the values logged to the console are the correct ones.

### Environment

Vaadin version(s): 23.3.9, 24.0.5
OS: Windows 10

### Browsers

Chrome

### PR DESCRIPTION
**Pull request by ugur-vaadin**
## Description

Currently, when `formatTime` in the time picker connector calls the helper function `formatMilliseconds`, the provided milliseconds value can be a string or a number. However, the helper function expects a number at this point and the calculations get broken. Therefore, the ms values smaller than 100 gets rounded down to 0. This PR makes sure that the helper function is only called with a number ms value.

Fixes #5101 

## Type of change

- [x] Bugfix
- [ ] Feature

## Checklist

- [x] I have read the contribution guide: https://vaadin.com/docs/latest/contributing/overview
- [x] I have added a description following the guideline.
- [x] The issue is created in the corresponding repository and I have referenced it.
- [x] I have added tests to ensure my change is effective and works as intended.
- [x] New and existing tests are passing locally with my change.
- [x] I have performed self-review and corrected misspellings.

### CODE REVIEW

File: vaadin-time-picker-flow-parent/vaadin-time-picker-flow/src/main/resources/META-INF/resources/frontend/vaadin-time-picker/timepickerConnector.js [hunk 1]
```diff
@@ -91,7 +91,11 @@ import {
 
               // milliseconds not part of the time format API
               if (includeMilliSeconds()) {
-                localeTimeString = formatMilliseconds(localeTimeString, timeObject.milliseconds, amString, pmString);
+                let timeObjectMilliseconds = timeObject.milliseconds;
+                if (typeof timeObjectMilliseconds === 'string') {
```
[the remark this answers is not in the document]
Comment (vursen) [reply vaadin-time-picker-flow-parent/vaadin-time-picker-flow/src/main/resources/META-INF/resources/frontend/vaadin-time-picker/timepickerConnector.js:95]: I decided to look into this deeper and found that `formatTime` actually receives a time object where all time parts are strings, not just the milliseconds. This happens only during initialization. The reason seems to be this part of the connector:

https://github.com/vaadin/flow-components/blob/27757de4c1eca6d5915b9804db92bfb1f8ac00c3/vaadin-time-picker-flow-parent/vaadin-time-picker-flow/src/main/resources/META-INF/resources/frontend/vaadin-time-picker/timepickerConnector.js#L39-L42

It calls `i18n.parseTime` before it's overridden, so the call goes to the default function of the web component, which returns an object with string values. I think it's not correct.

[the remark this answers is not in the document]
Comment (vursen) [reply vaadin-time-picker-flow-parent/vaadin-time-picker-flow/src/main/resources/META-INF/resources/frontend/vaadin-time-picker/timepickerConnector.js:95]: I think the logic is correct after all. When setting a new locale, the previous `i18n.parseTime` is used to be able to reformat the value with the new locale later.

[the remark this answers is not in the document]
Comment (vursen) [reply vaadin-time-picker-flow-parent/vaadin-time-picker-flow/src/main/resources/META-INF/resources/frontend/vaadin-time-picker/timepickerConnector.js]: What is inaccurate is that the connector uses the previous i18n parser for the value property, which is always in ISO format and does not depend on the selected locale. Although I haven't found a scenario where it causes an issue, it's more precise to use the web component's default i18n parser, which is ISO compliant. The fix then becomes naturally associated with that ISO parser.

**Review by vursen — APPROVED** [review]

File: vaadin-time-picker-flow-parent/vaadin-time-picker-flow/src/main/resources/META-INF/resources/frontend/vaadin-time-picker/timepickerConnector.js [hunk 2]
```diff
@@ -8,6 +8,7 @@ import {
   getSeparator,
   searchAmOrPmToken
 } from './helpers.js';
+import { TimePicker } from '@vaadin/time-picker';
```
Comment (web-padawan) [inline vaadin-time-picker-flow-parent/vaadin-time-picker-flow/src/main/resources/META-INF/resources/frontend/vaadin-time-picker/timepickerConnector.js:11]: I wonder if we should import from the `src` folder?

Comment (vursen) [reply vaadin-time-picker-flow-parent/vaadin-time-picker-flow/src/main/resources/META-INF/resources/frontend/vaadin-time-picker/timepickerConnector.js]: Updated.

**Review by DiegoCardoso — APPROVED** [review]

**Pull request merged**